crates/sparrow-transforms/src/transform_pipeline.rs

```diff
@@ -0,0 +1,174 @@
+use std::collections::VecDeque;
+use std::sync::Arc;
+
+use error_stack::ResultExt;
+use itertools::Itertools;
+use parking_lot::Mutex;
+use sparrow_arrow::Batch;
+use sparrow_scheduler::{
+    Partition, Partitioned, Pipeline, PipelineSink, Queue, Scheduler, Sink, TaskRef,
+};
+
+use crate::transforms::Transform;
+use crate::Error;
+
+/// Runs a linear sequence of transforms as a pipeline.
+pub struct TransformPipeline {
```

Comment:
TODO: Mention that transforms shouldn't change the partitioning of their batch (eg., they are effectively "map this sequence of transforms over the batches in this partition"

Reply:
?